spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -103,36 +108,36 @@ else if (this.tree.isDistinct()) {
 
 	@Override
 	public Object execute(Object[] parameters) {
-		List<T> results = executeRawResult(parameters);
-		if (this.tree.isCountProjection()) {
-			throw new DatastoreDataException(
-					"Count-queries are not natively supported for Cloud Datastore. "
-							+ "Please explicitly use a find-query and examine the result size.");
-		}
-		else if (this.tree.isExistsProjection()) {
-			return !results.isEmpty();
-		}
-		else {
-			return applyProjection(results);
-		}
+		List results = executeRawResult(parameters);
+		return ifCountExistsQuery(results::size, () -> !results.isEmpty(),
+				() -> applyProjection(results));
 	}
 
 	@Override
-	protected List<T> executeRawResult(Object[] parameters) {
-		Iterable<T> found = this.datastoreOperations.query(getQuery(parameters),
-				this.entityType);
+	protected List executeRawResult(Object[] parameters) {
```

Comment:
I'm not convinced you need this method at all. Why can't this logic just move into the method above?

Reply:
The reason is in this PR we are still not supporting count/exists queries for the query-string annotated query methods. There is then a difference in the `execute` method between part-tree and gql query method implementations. 

We have a separate issue for supporting count/exists queries, and I expect when implementing that one that executeRawResult and execute can be combined since there won't be a difference between GQL and part-tree query methods.